### ISSUE TITLE
Turn off a certain filter without deleting it

### ISSUE THREAD

**Issue by davesteps** [iterative/studio-support#21]
If you have a number of filters set up, sometimes it is nice to be able to turn off a certain filter but at the moment you need to remove it so you lose it completely. It would be nice to be able to disable the filter without deleting it. Thanks!

**Comment by Suor** [comment]
This might be less needed if we unify search and filter, i.e. if one is able to type in `branch=something` then erasing it and readding is way faster than clicking through the filter popup second time. Additionally we might show history and/or hints in a dropdown for the search field.

**Comment by mvshmakov** [comment]
I like the suggestion by Alex, at least GitHub does that, but the current way seems more visual to me. In addition, we would need a big search field in order to show at least a part of the filters in this case.

Against the initial suggestion by David, it seems pretty much a simple thing to do.

**Comment by Suor** [comment]
> but the current way seems more visual to me

We are not required to get rid of filters popup interface, any selections there might be duplicated into search string. Going with small steps is fine of cause, if it's easy to do then let's start with that.

**Comment by arcticbear** [comment]
What do you think if we just add the eye icons to each filter in the list to show/hide them? I hope this one not require complex changes. See in the example below ↓

<img width="1108" alt="image" src="https://user-images.githubusercontent.com/1275731/126331028-dbe19f5f-18c5-4dbf-ad79-6c1789e4086c.png">

**Comment by Suor** [comment]
The eye logic is kind of reverse here, i.e. if filter is active then we hide things. Maybe use some other icon to show that it's working/in effect/applied. Checkbox?

**Comment by jellebouwman** [comment]
Nice suggestion @arcticbear!

> The eye logic is kind of reverse here, i.e. if filter is active then we hide things. Maybe use some other icon to show that it's working/in effect/applied. Checkbox?

I'd say a toggle works better here. We are using checkboxes for multi selecting commits, and a toggle button for things like the ViewDelta mode 🤔

**Comment by arcticbear** [comment]
Yeah, I had been thinking about the checkboxes. And I like the idea of using toggles even more. Here are examples of how both options look like in the UI:

<img width="905" alt="image" src="https://user-images.githubusercontent.com/1275731/126352676-1602b5b0-c1e6-48bc-9d91-0d34e5bbe190.png">

**Comment by Suor** [comment]
Toggles look too heavy for me. Drain too much attention.

**Comment by arcticbear** [comment]
Yeah, agree. It was worth trying, but the checkboxes look much clear and non-distractive.

**Comment by shcheklein** [comment]
Toggles also don't contradict the idea about using the search bar and show filters GH style, right? Basically, we can combine best of two approaches - power search string _and_ a visual interface to populate it. Btw, I wonder how it's done on GitLab?
